core/common/src/main/java/alluxio/heartbeat/CronExpressionIntervalSupplier.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.heartbeat;
+
+import org.apache.logging.log4j.core.util.CronExpression;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+
+public class CronExpressionIntervalSupplier extends FixedIntervalSupplier {
```

Comment:
this shouldn't inherit from FixedIntervalSupplier. Cron intervals are not fixed.

Reply:
Fix it